healthcare/api-client/v1/datasets/datasets_test.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2018, 2019, 2020 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+import uuid
+
+from googleapiclient.errors import HttpError
+from retrying import retry
+
+import datasets
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+dataset_id = 'test-dataset-{}'.format(uuid.uuid4())
+destination_dataset_id = 'test-destination-dataset-{}'.format(uuid.uuid4())
+keeplist_tags = 'PatientID'
+time_zone = 'UTC'
+
+
+@pytest.fixture(scope="module")
+def test_dataset():
+    @retry(
+        wait_exponential_multiplier=1000,
+        wait_exponential_max=10000,
+        stop_max_attempt_number=10)
+    def create():
+        try:
+            datasets.create_dataset(project_id, cloud_region, dataset_id)
+        except HttpError as err:
```

Comment:
I think this except is never effective because `create_dataset` is swallowing all the exceptions.

My suggestion is to delete try-except from (at least) `create_dataset` and `delete_dataset`.

Those except clause is just printing failure and throwing away all the information about the failure.
I think it's better if the sample throws the `HttpError`.

Reply:
All try-except have been removed. Should we have 2 callbacks? One for 404s on delete and one for 409s on create?